uwuno/src/main/java/com/learning/uwuno/game/gameState.java

```diff
@@ -0,0 +1,151 @@
+        return turnsTaken;
+    }
+
+    public boolean isTurnDirection() {
+        return turnDirection;
+    }
+
+    public HashMap<player, Integer> getScores() {
+        return scores;
+    }
+
+    // To keep track of points and other current game properties
+    public gameState(playerList playerList) {
+        cardsToDraw = 0;
+        turnsTaken = 0;
+        turnDirection = true;
+        scores = new HashMap<>();
+
+        for (player player : playerList) {
+            scores.put(player, 0);
+        }
+    }
+
+    // For Point Mode: resets game state fields except scores
+    public void resetRound() {
+        cardsToDraw = 0;
+        turnsTaken = 0;
+        turnDirection = true;
+    }
+
+    public void applyCardEffect(card.CardType type) {
+        if (type == card.CardType.Draw2) {
+            cardsToDraw += 2;
+        } else if (type == card.CardType.Draw4) {
+            cardsToDraw += 4;
+        } else if (type == card.CardType.Reverse) {
+            turnDirection = !turnDirection;
+        }
+    }
+
+    public void skipTurn(player player, room room, gameResponse response) {
+        // Check if there are extra cards to draw from stacking effect
+        if (cardsToDraw > 0) {
+            ArrayList<card> playableCards = new ArrayList<>();
+            ArrayList<card> cards = player.drawCards(cardsToDraw);
+            cards.addAll(player.getCardList());
+            cardsToDraw = 0;
+            // If the card is valid to play, return response with current pid and playable cards
+            for (card card : cards) {
+                if (checkPlayable(card, room.lastPlayedCard())) {
+                    playableCards.add(card);
+                }
+            }
+            // Need to send response here because player has not taken their turn yet. Can still draw 1 card to skip.
+            response.setPlayerTurnResponse(player.getPid(), playableCards);
+            return;
+        }
+        ArrayList<card> cards = player.drawCards(1);
+        if (checkPlayable(cards.get(0), room.lastPlayedCard())) {
+            response.setPlayerTurnResponse(player.getPid(), cards);
+            // If card cannot be played, return response with next player's pid and their playable cards
+            endTurn(player, room, response);
+        }
+    }
+
+    public void endTurn(player player, room room, gameResponse response) {
+        ++turnsTaken;
+        gameSettings gameSettings = room.getGameSettings();
+        // If the given player has won, return winning response
+        if (player.getCardList().isEmpty() || turnsTaken == gameSettings.getMaxTurn()) {
+            ArrayList<player> winners = gameSettings.getGameMode().determineWinner(player, room.getPlayers(),
+                    scores, gameSettings.getMaxScore());
+            if (winners.isEmpty()) {
+                // No winners should only happen in Point Mode
+                // TODO: better way to trigger reset game state (except scores) for next round?
```

Comment:
In point mode, there can be multiple rounds. Right now, the way I am handling resetting the round while keeping the score is through PUT room request with Restart as the roomStatus.

Reply:
oh i see, thats very interesting